### PR TITLE
New package for a database layer

### PR DESCRIPTION
**Pull request by jaads**
## Description
On the way to isolate all database requests within a package, this PR adds the new package without doing anything yet. More PRs will come soon to make use of this new package.  

## Type of Change

- [ ] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [x] Other, please describe: Prerequisite for architectural improvement 

## Requirements Checklist

- [x] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

File: packages/datastore/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+	"name": "@directus/database-driver",
```
Comment (paescuj) [inline packages/datastore/package.json:2]:
```suggestion
	"name": "@directus/datastore",
```

File: packages/datastore/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		"dev": "tsc --watch",
+		"test": "vitest run",
+		"test:watch": "vitest",
+		"test:coverage": "vitest run --coverage"
+	},
+	"description": "Database layer for Directus.",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/directus/directus.git",
+		"directory": "packages/persistance"
```
Comment (paescuj) [inline packages/datastore/package.json:16]:
```suggestion
		"directory": "packages/datastore"
```

File: packages/datastore/readme.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# `@directus/database-driver`
```
Comment (paescuj) [inline packages/datastore/readme.md:1]:
```suggestion
# `@directus/datastore`
```

File: packages/datastore/readme.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+
+Database layer for `@directus/directus`. All database calls will go trough this layer in the near feature.
```
Comment (paescuj) [inline packages/datastore/readme.md:3]:
```suggestion
Database layer for `@directus/directus`. All database calls will go through this layer in the near feature.
```

**Comment by jaads** [comment]
Oh wow thanks @paescuj!

**Comment by paescuj** [comment]
> Oh wow thanks @paescuj!

You're very welcome! ❤️

File: packages/datastore/package.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	"version": "0.0.1",
+	"type": "module",
+	"scripts": {
+		"build": "tsc --build",
+		"dev": "tsc --watch",
+		"test": "vitest run",
+		"test:watch": "vitest",
+		"test:coverage": "vitest run --coverage"
+	},
+	"description": "Database layer for Directus.",
```
Comment (paescuj) [inline packages/datastore/package.json:12]: Sorry for nitpicking! 🙈 But I remember that we've removed the periods from the description in all packages some time ago... 😃 

```suggestion
	"description": "Database layer for Directus",
```

Comment (jaads) [reply packages/datastore/package.json]: No problem, I didn't know :+1:

**Comment by jaads** [comment]
Closed this PR as we're thinking about another approach to realize the mentioned database layer.

**Pull request closed without merging**